test/ng/rootScopeSpec.js

```diff
@@ -2803,6 +2803,45 @@ describe('Scope', function() {
         }));
       });
     });
+
+
+    it('should allow recursive $emit/$broadcast', inject(function($rootScope) {
+      var callCount = 0;
+      $rootScope.$on('evt', function($event, arg0) {
+        callCount++;
+        if (arg0 !== 1234) {
+          $rootScope.$emit('evt', 1234);
+          $rootScope.$broadcast('evt', 1234);
+        }
+      });
+
+      $rootScope.$emit('evt');
+      $rootScope.$broadcast('evt');
+      expect(callCount).toBe(6);
+    }));
+
+
+    it('should allow recursive $emit/$broadcast between parent/child', inject(function($rootScope) {
```

Comment:
I don't think this test demonstrates recursive emitting/broadcasting.

Reply:
How so? It does at least one recursive `$broadcast` + one `$emit` within `$broadcast`...

I'm sure we could think of better or more interesting tests though. Let me know if you have any specific ones you think should be added.

Reply:
Yeah, it does `root.$broadcast` inside `root.$broadcast` (which we don't care about in this test) and `child.$emit` inside `root.$broadcast`, so in a way it is recursive. But it doesn't test for `root.$broadcast` inside `child.$emit`, for example.

In order to test both `root.$broadcast --> child.$emit` and `child.$emit --> root.$broadcast`, I would change it to something like:

```js
var child = $rootScope.$new();
var calls = '';

$rootScope.$on('evt', function($event, arg0) {
  calls += 'r';  // For "root".
  if (arg0 === 'fromChild') {
    $rootScope.$broadcast('evt', 'fromRoot2');
  }
});

child.$on('evt', function($event, arg0) {
  calls += 'c';  // For "child".
  if (arg0 === 'fromRoot1') {
    child.$emit('evt', 'fromChild');
  }
});

$rootScope.$broadcast('evt', 'fromRoot1');
expect(calls).toBe('rccrrc');
```

Reply:
I updated to use your test